src/main/c/powerwaf_jni.c

```diff
@@ -549,7 +571,18 @@ static jobject _run_rule_common(bool is_byte_buffer, JNIEnv *env, jclass clazz,
         _throw_pwaf_exception(env, DDWAF_ERR_INTERNAL);
         goto end;
     }
-    DDWAF_RET_CODE ret_code = ddwaf_run(ctx, &input, &ret, run_budget);
+
+    ddwaf_object * persistent_input_ptr = NULL;
+    if (persistent_input.type != DDWAF_OBJ_NULL) {
+        persistent_input_ptr = &persistent_input;
+    }
+
+    ddwaf_object * ephemeral_input_ptr = NULL;
+    if (ephemeral_input.type != DDWAF_OBJ_NULL) {
+        ephemeral_input_ptr = &ephemeral_input;
+    }
+
+    DDWAF_RET_CODE ret_code = ddwaf_run(ctx, persistent_input_ptr, ephemeral_input_ptr, &ret, run_budget);
```

Comment:
you should check if both persistent and ephemeral are null, because then it's an invalid call

Reply:
Added extra check